source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -474,21 +475,23 @@ namespace
 
 } // end anonymous namespace
 
-RtCoreIo::RtCoreIo(RtObject stage) :
+RtFileIo::RtFileIo(RtObject stage) :
     RtApiBase(stage)
 {
 }
 
-RtApiType RtCoreIo::getApiType() const
+RtApiType RtFileIo::getApiType() const
 {
     return RtApiType::CORE_IO;
 }
 
-void RtCoreIo::read(const DocumentPtr& doc, RtCoreIo::ReadFilter filter)
+void RtFileIo::read(const DocumentPtr& doc, RtFileIo::ReadFilter filter)
 {
     PrvStage* stage = data()->asA<PrvStage>();
+    stage->clearAttributes();
     readAttributes(doc, stage, {});
 
+    stage->clearElements();
```

Comment:
Clearing the stage before read could be done on client side instead. Then this read method would be used for both reading from scratch and reading to append to existing stage contents.

Reply:
But they need to clear attributes + elements. I guess I could add a RtStage::initialize(). Like there is for Document.